### DISCUSSION TITLE
Harmonic Number

### DISCUSSION THREAD

**Discussion by Bumblebee001** [GrandOrgue/GoOdf#24] [General]
In GrandOrgue a 16' rank has a harmonic number of 4; a 4' rank has a HN of 16..... and so on.

What number should be entered for Harmonic number? Could you have a drop down menu with the appropriate listing of different pitches with their corresponding harmonic numbers to facilitate selection especially when the mixtures and mutation stops are concerned?

**Comment by larspalo** [comment]
The calculation of what the HarmonicNumber should be is done by taking 64 and divide it by the foot number. This extends to any strange or "uncommon" rank sizes. Like the "Septima 1 1/7'" would get a HarmonicNumber of 56 which is calculated by 64 / (1 + (1/7)). In the same way, a "Quinta 10 2/3'" would have HarmonicNumber of 6, which is calculated by 64 / (10 + (2 / 3)).

Using a spinctrl allows for any integer value (1 to 1024 for GO), whereas a dropdown would limit to only what choices are "allowed" in it. Also the dropdown would potentially be slower than just writing the value in the spinctrl and tabbing out of it to make it take effect.

As this is not quite intuitive what number to write, I however think that some kind of help with it would indeed be beneficial - I just have to think about how to best implement it... (perhaps a statictext explaining what the resulting foot length would be for the current number entered)

**Reply by larspalo** [reply]
I've implemented a wxStaticText as a help text that will show what pitch foot length is equivalent to the specified HarmonicNumber in the wxSpinCtrl. This allows for all the valid values to be entered, while providing a hint to the user what the "pitch value" of the number actually is. A new release, containing this and a few other fixes, is building as I write this message.

**Comment by Bumblebee001** [comment]
Looking forward to this and thanks for your work!! It is immensely
appreciated.

**Comment by Bumblebee001** [comment]
I have installed the update.

I might add a few more details:
Middle C = C3:

10' 2/3      G2  (Quint Bass)
8'                C3  (Prestant, Principal, Prinzipal(e))
5' 1/3         G3  (Quint)
*4'                C4  (1 Octave)*
3' 1/5         E4  (Gross Tierce)
2' 2/3         G4  (Nazard)
*2'                 C5  (2 Octave)*
1' 3/5          E5  (Tierce, Terz)
1' 1/3          G5  (Larigot)
1' 1/7          B5  (Septieme)*
**1'                 C6  (3 Octave)*
8/9              D6  (None)
8/13            A6  (Tredezime)
*1/2               C7  (4 Octave)*
8/19             Eb7(Molterz)
*1/4               C8  (5 Octave)*
1/6               G8  (Quadragesima)

I included the respective key/ note to give meaning to the terms and 
fractions. I drew up a chart some time ago and I refer to it now and 
again and helps me get a sense of orientation. You may know more on this 
than I do I'm sure. Here it is:



Mark

**Comment by Bumblebee001** [comment]
In the case of mixtures, the harmonic number is not only difficult to decide but experience revealed that the harmonic number needs to be progressively changed since the rank breaks at various stages up the scale. How will this translate to this software?

**Reply by larspalo** [reply]
For now, you have to edit the HarmonicNumber on the pipe level manually. Set the base HarmonicNumber first for the rank/stop and then at the first change (break back) select that pipe and edit it's properties (Ctrl+E, or right click and select it in menu). In that dialog it's possible to step forwards (and backwards) and alter (or even better - if you know how many pipes to alter, use the copy properties functionality of that dialog to copy to a number of following pipes.

I'm however considering adding an auto detect feature for the HarmonicNumber that would depend on the embedded pitch info in the samples. This however requires that I expand the WAVfileParser capability to include parsing the smpl chunk of each pipe (first attack sample)... It's not a very high priority though.
